gen/generator.go

```diff
@@ -823,10 +950,61 @@ func main() {
 			renderTemplate("datasource.md.tmpl", fmt.Sprintf("%s.md", model.ResourceName), datasourcesDocsPath, model)
 			renderTemplate("resource_test.go.tmpl", fmt.Sprintf("resource_%s_%s_test.go", providerName, model.ResourceName), providerPath, model)
 			renderTemplate("datasource_test.go.tmpl", fmt.Sprintf("data_source_%s_%s_test.go", providerName, model.ResourceName), providerPath, model)
-
 		}
 	}
+}
+
+type Version struct {
+	Major int
+	Minor int
+	Patch int
+	Tag   int
+}
 
+type VersionResult struct {
+	Version *Version
+	Error   string
+}
+
+func parseVersion(rawVersion string) VersionResult {
+	versionRegex := regexp.MustCompile(`(\d+)\.(\d+)\((\d+)([a-z])\)`)
+	matches := versionRegex.FindStringSubmatch(rawVersion)
+	if matches == nil {
+		return VersionResult{Error: "unknown"}
+	}
+	major, _ := strconv.Atoi(matches[1])
+	minor, _ := strconv.Atoi(matches[2])
+	patch, _ := strconv.Atoi(matches[3])
+	tag := int(matches[4][0])
+
+	return VersionResult{Version: &Version{Major: major, Minor: minor, Patch: patch, Tag: tag}}
+
+}
+
+func isVersionGreater(v1, v2 Version) bool {
+	if v1.Major != v2.Major {
+		return v1.Major > v2.Major
+	}
+	if v1.Minor != v2.Minor {
+		return v1.Minor > v2.Minor
+	}
+	if v1.Patch != v2.Patch {
+		return v1.Patch > v2.Patch
+	}
+	return v1.Tag > v2.Tag
+}
+
+func isVersionLesser(v1, v2 Version) bool {
+	if v1.Major != v2.Major {
+		return v1.Major < v2.Major
+	}
+	if v1.Minor != v2.Minor {
+		return v1.Minor < v2.Minor
+	}
+	if v1.Patch != v2.Patch {
+		return v1.Patch < v2.Patch
+	}
+	return v1.Tag < v2.Tag
 }
```

Comment:
Remove this as it will be unnecessary when we move the functions in the provider directory/package

Reply:
ok